transport/transporters/kafka/factory.go

```diff
@@ -90,16 +90,16 @@ func New(
 
 	config, _ := producerConfig(tls, ca, privateKey, publicKey, maxMessageBytes)
 	bootstrapServer := fmt.Sprintf("%s:%s", kafkaHost, kafkaPort)
-	syncProducer, _ := sarama.NewSyncProducer([]string{bootstrapServer}, config)
-	if verifyProducer {
-		if err := verifySend(&syncProducer, topic); err != nil {
-			panic(err)
-		}
-	}
 
 	transports := make([]*transport.Transporter, workers)
 
 	for i := 0; i < workers; i++ {
+		syncProducer, _ := sarama.NewSyncProducer([]string{bootstrapServer}, config)
+		if verifyProducer {
+			if err := verifySend(&syncProducer, topic); err != nil {
+				panic(err)
+			}
+		}
```

Comment:
I'm choosing to create a producer per worker. 

I thought about the case when two different workers are writing two different batches of transactions to the same partition. If the workers share a kafka producer, there's a high chance messages within both transactions become intermingled with each other in the downstream kafka buffers.

To prevent this, if each worker has it's own producer, they'll each have their own downstream sarama buffers where batches are written to kafka from.